### ISSUE TITLE
Using Text as Map key is impossible

### ISSUE THREAD

**Issue by krzyk** [yegor256/cactoos#788]
Right now when I do:
```
        final String KEY = "test";
        final Integer found = new MapOf<>(
            new MapEntry<>(new TextOf(KEY), 1)
        ).get(new TextOf(KEY));
        System.out.println(found);
```
I get a `null` because `TextOf` doesn't implement hashcode and equals.

To fix this create a decorator (I'm not sure about the name, EqualsHashCodeText seams to long) that would get the string from `asString` and use its `hashCode` and equals method in this decorators `hashCode` and equals.

**Comment by llorllale** [comment]
@krzyk let's create a `TextEnvelope` that implements hashcode and equals for all other texts.

**Comment by 0crat** [comment]
Job [#788](https://github.com/yegor256/cactoos/issues/788) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
The job [#788](https://github.com/yegor256/cactoos/issues/788) assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C63314D6Z/1251917370); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)

**Comment by 0pdd** [comment]
@krzyk the puzzle [#849](https://github.com/yegor256/cactoos/issues/849) is still not solved.

**Comment by krzyk** [comment]
@paulodamaso thanks

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#788](https://github.com/yegor256/cactoos/issues/788) is now out of scope

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Comment by 0pdd** [comment]
@krzyk the puzzle [#878](https://github.com/yegor256/cactoos/issues/878) is still not solved; solved: [#849](https://github.com/yegor256/cactoos/issues/849).

**Comment by 0pdd** [comment]
@krzyk the puzzle [#897](https://github.com/yegor256/cactoos/issues/897) is still not solved; solved: [#849](https://github.com/yegor256/cactoos/issues/849), [#878](https://github.com/yegor256/cactoos/issues/878).

**Comment by 0pdd** [comment]
@krzyk the puzzle [#929](https://github.com/yegor256/cactoos/issues/929) is still not solved; solved: [#849](https://github.com/yegor256/cactoos/issues/849), [#878](https://github.com/yegor256/cactoos/issues/878), [#897](https://github.com/yegor256/cactoos/issues/897).

**Comment by 0pdd** [comment]
@krzyk the puzzle [#1063](https://github.com/yegor256/cactoos/issues/1063) is still not solved; solved: [#849](https://github.com/yegor256/cactoos/issues/849), [#878](https://github.com/yegor256/cactoos/issues/878), [#897](https://github.com/yegor256/cactoos/issues/897), [#929](https://github.com/yegor256/cactoos/issues/929).

**Comment by 0pdd** [comment]
@krzyk the puzzle [#1116](https://github.com/yegor256/cactoos/issues/1116) is still not solved; solved: [#1063](https://github.com/yegor256/cactoos/issues/1063), [#849](https://github.com/yegor256/cactoos/issues/849), [#878](https://github.com/yegor256/cactoos/issues/878), [#897](https://github.com/yegor256/cactoos/issues/897), [#929](https://github.com/yegor256/cactoos/issues/929).

**Comment by 0pdd** [comment]
@krzyk the puzzle [#1135](https://github.com/yegor256/cactoos/issues/1135) is still not solved; solved: [#1063](https://github.com/yegor256/cactoos/issues/1063), [#1116](https://github.com/yegor256/cactoos/issues/1116), [#849](https://github.com/yegor256/cactoos/issues/849), [#878](https://github.com/yegor256/cactoos/issues/878), [#897](https://github.com/yegor256/cactoos/issues/897), [#929](https://github.com/yegor256/cactoos/issues/929).

**Comment by 0pdd** [comment]
@krzyk 2 puzzles [#1135](https://github.com/yegor256/cactoos/issues/1135), [#1261](https://github.com/yegor256/cactoos/issues/1261) are still not solved; solved: [#1063](https://github.com/yegor256/cactoos/issues/1063), [#1116](https://github.com/yegor256/cactoos/issues/1116), [#849](https://github.com/yegor256/cactoos/issues/849), [#878](https://github.com/yegor256/cactoos/issues/878), [#897](https://github.com/yegor256/cactoos/issues/897), [#929](https://github.com/yegor256/cactoos/issues/929).

**Comment by 0pdd** [comment]
@krzyk the puzzle [#1261](https://github.com/yegor256/cactoos/issues/1261) is still not solved; solved: [#1063](https://github.com/yegor256/cactoos/issues/1063), [#1116](https://github.com/yegor256/cactoos/issues/1116), [#1135](https://github.com/yegor256/cactoos/issues/1135), [#849](https://github.com/yegor256/cactoos/issues/849), [#878](https://github.com/yegor256/cactoos/issues/878), [#897](https://github.com/yegor256/cactoos/issues/897), [#929](https://github.com/yegor256/cactoos/issues/929).

**Comment by 0pdd** [comment]
@krzyk all 8 puzzles are solved here: [#1063](https://github.com/yegor256/cactoos/issues/1063), [#1116](https://github.com/yegor256/cactoos/issues/1116), [#1135](https://github.com/yegor256/cactoos/issues/1135), [#1261](https://github.com/yegor256/cactoos/issues/1261), [#849](https://github.com/yegor256/cactoos/issues/849), [#878](https://github.com/yegor256/cactoos/issues/878), [#897](https://github.com/yegor256/cactoos/issues/897), [#929](https://github.com/yegor256/cactoos/issues/929).
